### PR TITLE
feat: add dataset.workspace.files for provisioning case source files

### PR DESCRIPTION
**Pull request by StevenTobin**
## Description

Skills that operate on source code (e.g., autofix) need actual files in the workspace, not just input.yaml metadata. Add a dataset.input_files_dir config field (default: files). When a case directory contains a subdirectory with that name, its contents are recursively copied into the workspace root. The directory name is added to _SKIP_NAMES so it is not treated as input data in batch mode. No breaking changes -- cases without a files/ directory are unaffected.

## How Has This Been Tested?

Using autofix-skills as a testbed.

```
Yes, confirmed! The code change is working correctly:

  Dataset structure:
  - Files are in eval/cases/case-001-simple-null-pointer-fix/files/ subdirectory ✓

  Workspace structure:
  - Files copied to workspace root /tmp/agent-eval/2026-05-21-opus-v2/cases/case-001-simple-null-pointer-fix/ ✓
  - user_profile.py and test_user_profile.py are directly accessible ✓

  Execution:
  - Skill is successfully reading the files (logs show "Reading: implement-agent.md") ✓
  - No file-not-found errors ✓

  The updated harness is correctly loading files from the files/ subdirectory (as specified by input_files_dir: files in eval.yaml line 48) and copying them
  to the workspace root where the skill expects them.
```

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

## Summary by CodeRabbit

* **New Features**
  * Dataset settings are grouped under a single dataset section in config.
  * Config can list workspace input files; listed inputs are copied into evaluation workspaces, preserving nested structure.

* **Bug Fixes**
  * Workspace file entries validated (must be relative strings, no parent traversal).
  * Symlink escapes prevented; missing entries are treated as no-ops.

* **Tests**
  * Added comprehensive tests for parsing, copying behavior, and security edge cases.

### CODE REVIEW

File: skills/eval-run/scripts/workspace.py [hunk 1]
```diff
@@ -281,7 +290,7 @@ def _read_input(case_dir):
     Prefers files named 'input.*', then falls back to first parseable file.
     Skips known non-input files like answers.yaml and reference.*.
     """
-    _SKIP_NAMES = {"answers", "reference", "expected", "gold"}
+    _SKIP_NAMES = {"answers", "reference", "expected", "gold", "files"}
```
Comment (astefanutti) [inline skills/eval-run/scripts/workspace.py]: That should use the config value.

File: agent_eval/config.py [hunk 2]
```diff
@@ -217,6 +217,7 @@ class EvalConfig:
     # Dataset — natural language schema + path
     dataset_path: str = ""
     dataset_schema: str = ""
+    dataset_input_files_dir: str = "files"
```
Comment (astefanutti) [inline agent_eval/config.py]: Maybe we can start grouping those into `DatasetConfig`?

**Review by astefanutti — COMMENTED** [review]
@StevenTobin thanks, that looks good overall, just on the config structure to be more evolutive, I would suggest to have a `.dataset.workspace` field with the `files` sub-field, like:

```
  dataset:                                                                                                                                                                                                                                                                                                        
    workspace:                                                                                                                                                                                                                                                                                                    
      files:                                                                                                                                                                                                                                                                                                      
        - src/    
        - tickets/JIRA-123.md
        - config/settings.json                                                                                                                                                                                                                                                                                    
```
                                                                                                                                                                                                                                                                                                        
  Case directory:
```
  case-001/                                                                                                                                                                                                                                                                                                       
    input.yaml    
    annotations.yaml                                                                                                                                                                                                                                                                                              
    src/            
      main.py                                                                                                                                                                                                                                                                                                     
      utils.py    
    tickets/  
      JIRA-123.md
    config/      
      settings.json                                                                                                                                                                                                                                                                                               
      secrets.json        # NOT copied — not in the list
    reference.md
```

  Workspace after copy:
```
  /tmp/agent-eval/{id}/cases/case-001/                                                                                                                                                                                                                                                                            
    input.yaml            # copied by existing logic
    src/                                                                                                                                                                                                                                                                                                          
      main.py             # from workspace.files: src/                                                                                                                                                                                                                                                            
      utils.py                                                                                                                                                                                                                                                                                                    
    tickets/                                                                                                                                                                                                                                                                                                      
      JIRA-123.md         # from workspace.files: tickets/JIRA-123.md                                                                                                                                                                                                                                             
    config/                                                                                                                                                                                                                                                                                                       
      settings.json       # from workspace.files: config/settings.json                                                                                                                                                                                                                                            
                          # secrets.json NOT here — not listed        
    .claude/                                                                                                                                                                                                                                                                                                      
      settings.json       # created by workspace.py                                                                                                                                                                                                                                                               
    scripts/ → symlink
```

**Comment by rhuss** [comment]
Hey @StevenTobin, are you still actively working on this PR? There are two small items from @astefanutti's review to address plus a rebase onto current main.

If you're busy with other things, no worries at all! I can take it over and get it across the finish line, as we need this feature for some current work.

**Comment by StevenTobin** [comment]
@rhuss Thanks for the poke, no I can get this updated today, it is 100% on my list.

**Review by astefanutti — APPROVED** [review]
Thanks @StevenTobin!
